### PR TITLE
[issue-12] set an empty json if nor request payload was found

### PR DESCRIPTION
**Pull request by chaouchAbderraouf**
https://github.com/evaisse/SimpleHttpBundle/issues/12

### CODE REVIEW

**Comment by lahaxearnaud** [comment]
You broke some tests 🤕

File: Http/Statement.php [hunk 1]
```diff
@@ -239,7 +239,12 @@ public function json($json = null)
         $this->request->headers->set('charset', 'utf-8');
         $this->request->headers->set('accept', 'application/json');
         if ($this->request->getMethod() !== "GET") {
-            $json = $json === null ? json_encode($this->request->request->all()) : (string)$json;
+            if ($json === null && !empty($this->request->request->all())) {
```
Comment (lahaxearnaud) [inline Http/Statement.php]: Store $this->request->request->all() in a var

Comment (chaouchAbderraouf) [reply Http/Statement.php]: fixed

File: Http/Statement.php [hunk 2]
```diff
@@ -239,9 +239,20 @@ public function json($json = null)
         $this->request->headers->set('charset', 'utf-8');
         $this->request->headers->set('accept', 'application/json');
         if ($this->request->getMethod() !== "GET") {
-            $json = $json === null ? json_encode($this->request->request->all()) : (string)$json;
+            $requestPayload = $this->request->request->all();
+            if(func_num_args() === 0) {
```
Comment (evaisse) [inline Http/Statement.php]: conventions `if (`

Comment (chaouchAbderraouf) [reply Http/Statement.php]: Fixed

**Comment by evaisse** [comment]
Tip top merci

**Pull request merged**